openstack/identity/v3/roles/requests.go

```diff
@@ -148,3 +148,74 @@ func ListAssignments(client *gophercloud.ServiceClient, opts ListAssignmentsOpts
 		return RoleAssignmentPage{pagination.LinkedPageBase{PageResult: r}}
 	})
 }
+
+// AssignOpts provides options to assign a role
+type AssignOpts struct {
+	// UserID is the ID of a user to assign a role
+	// Note: exactly one of UserID or GroupID must be provided
+	UserID string `xor:"GroupID"`
+
+	// GroupID is the ID of a group to assign a role
+	// Note: exactly one of UserID or GroupID must be provided
+	GroupID string `xor:"UserID"`
+
+	// ProjectID is the ID of a project to assign a role on
+	// Note: exactly one of ProjectID or DomainID must be provided
+	ProjectID string `xor:"DomainID"`
+
+	// DomainID is the ID of a domain to assign a role on
+	// Note: exactly one of ProjectID or DomainID must be provided
+	DomainID string `xor:"ProjectID"`
+}
+
+func toAssignRoleURL(client *gophercloud.ServiceClient, roleID string, opts AssignOpts) (string, error) {
```

Comment:
If this function is kept, it should be renamed to `toRoleAssignURL`. Trivial change, but it follows the naming pattern throughout Gophercloud.